### PR TITLE
fix(interpolation): query key is out of range due to double calculation error

### PR DESCRIPTION
**Pull request by takayuki5168**
## Description

This range check sometimes fails even in a normal case due to calculation error of double, resulting in the node dying.
https://github.com/autowarefoundation/autoware.universe/blob/d10310ddbbf5aaba20858173cb26e7fdcb30ea49/common/interpolation/include/interpolation/interpolation_utils.hpp#L74

To fix this issue, I created this PR.
If the query key's violation from the range of base_keys is epsilon, it is judged as valid and cropped to be inside the range of base_keys.

TIER IV internal link
https://tier4.atlassian.net/browse/T4PB-22737
## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by takayuki5168** [comment]
@tkimura4 @rej55 Could you review please.
I confirmed that failed scenario passed with this PR

File: common/interpolation/include/interpolation/interpolation_utils.hpp [hunk 1]
```diff
@@ -71,9 +72,20 @@ inline void validateKeys(
...
-  if (query_keys.front() < base_keys.front() || base_keys.back() < query_keys.back()) {
+  constexpr double epsilon = 1e-3;
+  if (
+    query_keys.front() < base_keys.front() - epsilon ||
+    base_keys.back() + epsilon < query_keys.back()) {
     throw std::invalid_argument("query_keys is out of base_keys");
   }
+
+  // NOTE: Due to calculation error of double, a query key may be slightly out of base keys.
+  //       Therefore, query keys are cropped here.
```
Comment (tkimura4) [inline common/interpolation/include/interpolation/interpolation_utils.hpp:83]: :+1:

**Review by tkimura4 — APPROVED** [review]
LGTM

**Review by rej55 — APPROVED** [review]
LGTM

**Pull request merged**